### PR TITLE
Robert Rak -- Sprint-Challenge---React -- WIP

### PR DESCRIPTION
**Pull request by rrak22**

### CODE REVIEW

File: Answers.md [hunk 1]
```diff
@@ -1,9 +1,29 @@
 # Answers
 
-1. What is React JS? How does it differ from other JavaScript frameworks?
+## 1. What is React JS? How does it differ from other JavaScript frameworks?
 
+React JS is a framework for building web applications that can dynamically update their content without disrupting the user experience, the end result is that they feel more like desktop applications using web technologies. One way it differs from other frameworks is its use of a Virtual DOM, allowing it to very cheaply create elements on the fly (which end up just being plain objects) that are then translated to real HTML elements later. This ends up becoming part of the React Component Lifecycle, a feature that differentiates it from other frameworks which do not re-render components on state changes.
```
Comment (lefrenk) [inline Answers.md]: Well done!

File: starwars/src/CharacterList.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import CharacterCard from './CharacterCard';
+import './CharacterList.css';
+
+const CharacterList = props => {
```
Comment (lefrenk) [inline starwars/src/CharacterList.js]: You can use object destructuring here:

```
const CharacterList = ({ characters  }) => {

...

// line 8
  {characters.map(character => {
```

File: starwars/src/CharacterCard.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Card, CardTitle, CardBody, CardText, CardHeader, CardFooter } from 'reactstrap';
+import './CharacterCard.css';
+
+const CharacterCard = props => {
+  return (
+    <div className="CharacterCard">
+      <Card>
+          <CardHeader>
```
Comment (lefrenk) [inline starwars/src/CharacterCard.js]: Indentation is off here.

File: starwars/src/CharacterCard.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+          <CardTitle>{props.character.name}</CardTitle>
+        </CardHeader>
+        <CardBody>
+        <CardText>Height: {props.character.height}</CardText>
```
Comment (lefrenk) [inline starwars/src/CharacterCard.js]: `<CardText>` should be indented since they're children of `<CardBody>`

**Comment by lefrenk** [comment]
Solid work, Robert! You've clearly demonstrated that you _get_ React. Keep up the good work :)

Please close this PR after reviewing feedback.

File: starwars/src/App.js [hunk 5]
```diff
@@ -27,6 +28,7 @@ class App extends Component {
     return (
       <div className="App">
         <h1 className="Header">React Wars</h1>
+        <CharacterList characters={this.state.starwarsChars}/>
```
Comment (JasonCorp84) [inline starwars/src/App.js]: 👍 
I can tell you can pass data via props

File: starwars/src/CharacterCard.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  return (
+    <div className="CharacterCard">
+      {console.log(props)}
+      <p>Name: {props.character.name}</p>
+      <p>Height: {props.character.height}</p>
+      <p>Mass: {props.character.mass}</p>
+      <p>Hair color: {props.character.hair_color}</p>
+      <p>Skin color: {props.character.skin_color}</p>
+    </div>
+  )
```
Comment (JasonCorp84) [inline starwars/src/CharacterCard.js]: 👍

File: starwars/src/CharacterList.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import CharacterCard from './CharacterCard';
+//import './CharacterList.css';
+
+const CharacterList = props => {
+  return (
+    <div className="CharacterListContainer">
+    {props.characters.map(character => {
+      return <CharacterCard key={character.name} character={character} />;
+    })}
```
Comment (JasonCorp84) [inline starwars/src/CharacterList.js]: 👍 
Nice option with the character name as a key. // since it is really unique

**Pull request closed without merging**